Mathlib/FieldTheory/Differential/Liouville.lean

```diff
@@ -0,0 +1,207 @@
+    -- logDeriv turns sums to products, so the new `u` will be the product of the old `u` over all
+    -- isomorphisms
+    let u'' (i : ι) := ∏ x : (K ≃ₐ[F] K), x (u i)
+
+    -- Each of the values of u'' are fixed by all isomorphisms.
+    have : ∀ i, u'' i ∈ fixedField (⊤ : Subgroup (K ≃ₐ[F] K)) := by
+      rintro i ⟨e, _⟩
+      change e (u'' i) = u'' i
+      simp only [u'', map_prod]
+      apply Fintype.prod_equiv (Equiv.mulLeft e)
+      simp
+    have ffb : fixedField ⊤ = ⊥ := (IsGalois.tfae.out 0 1).mp (inferInstanceAs (IsGalois F K))
+    simp_rw [ffb, IntermediateField.mem_bot, Set.mem_range] at this
+
+    -- Therefore they are all in `F`. We use `choose` to get their values in `F`.
+    choose u' hu' using this
+
+    -- We do almost the same thing for `v''`, just with sum instead of product.
+    let v'' := (∑ x : (K ≃ₐ[F] K), x v) / (Fintype.card ((K ≃ₐ[F] K)))
+    have : v'' ∈ fixedField (⊤ : Subgroup (K ≃ₐ[F] K)) := by
+      rintro ⟨e, _⟩
+      change e v'' = v''
+      simp only [v'', map_div₀, map_sum, map_natCast]
+      congr 1
+      apply Fintype.sum_equiv (Equiv.mulLeft e)
+      simp
+    rw [ffb, IntermediateField.mem_bot] at this
+    obtain ⟨v', hv'⟩ := this
+
+
+    exists ι, inferInstance, c', ?_, u', v'
+    · -- We need to prove that all `c'` are constants.
+      -- This is true because they are the division of a constant by
+      -- a natural nubmer (which is also constant)
+      intro x
+      simp [c', Derivation.leibniz_div, hc]
+    · -- Proving that this works is mostly straightforward algebraic manipulation,
+      apply_fun (algebraMap F K)
+      case inj =>
+        exact NoZeroSMulDivisors.algebraMap_injective F K
+      simp only [map_add, map_sum, map_mul, ← logDeriv_algebraMap, hu', ← deriv_algebraMap, hv']
+      unfold u'' v'' c'
+      clear c' u'' u' hu' v'' v' hv'
+      push_cast
+      rw [Derivation.leibniz_div_const, smul_eq_mul, inv_mul_eq_div]
+      case h => simp
+      simp only [map_sum, div_mul_eq_mul_div]
+      rw [← sum_div, ← add_div]
+      field_simp
+      -- Here we rewrite logDeriv (∏ x : K ≃ₐ[F] K, x (u i)) to ∑ x : K ≃ₐ[F] K, logDeriv (x (u i))
+      conv =>
+        enter [2, 1, 2, i, 2]
+        equals ∑ x : K ≃ₐ[F] K, logDeriv (x (u i)) =>
+          by_cases h : u i = 0 <;>
+          simp [logDeriv_prod_of_eq_zero, logDeriv_prod, h]
+      simp_rw [mul_sum]
+      rw [sum_comm, ← sum_add_distrib]
+      trans ∑ _ : (K ≃ₐ[F] K), a
+      · simp [mul_comm]
+      · rcongr e
+        apply_fun e at h
+        simp only [AlgEquiv.commutes, map_add, map_sum, map_mul] at h
+        convert h using 2
+        · rcongr x
+          simp [logDeriv, algEquiv_deriv']
+        · rw [algEquiv_deriv']
+
+/--
+We lift `isLiouville_of_finiteDimensional_galois` to non-Galois field extensions by using it for the
+normal closure then obtaining it for `F`.
+-/
+instance liouville_of_finiteDimensional [FiniteDimensional F K] :
```

Comment:
I think this result deserves being mentioned under `## Main declarations` in the module docstring.

Reply:
Did you forget to push? Because I don't see any changes to the module docstring...

Reply:
I see them at https://github.com/leanprover-community/mathlib4/pull/16797/files#diff-3d46a76fa9cf3594c6cc2d1d5c71943343f0803751417cede4f798105ce755a6R25

Reply:
Yes, now I see them. I don't know what happened...